javascript/javascript2/week1/Javascript2Project/index.js

```diff
@@ -0,0 +1,108 @@
+let gameSize = 0;
+
+const cardId = 0;
+const img = document.createElement("img");
+const cardDiv = document.getElementById("cards-grid");
+const cardStorageArray = [];
+const randomCardsArray = [];
+const backSrc = "./cards/cardBack.webp";
+const cardsArray = [];
+const suits = ["Hearts", "Diamonds", "Clubs", "Spades"];
+const deck = ["Two", "Three", "Four", "Five", "Six", "Seven", "Eight", "Nine", "Ten", "Jack", "Queen", "King", "Ace"];
+
+const suitsFirstLetter = suits.map((firstLetter) => {
+  return firstLetter[0];
+});
+
+function choseDifficulty() {
+  if (!gameSize) {
+    let input = prompt("Please enter the amount of pairs you want to play with!", "8");
+
+    if (input !== null && input.trim() !== "" && !isNaN(input) && Number(input) > 0) {
+      gameSize = Number(input);
+      document.getElementById("landing-text").style.display = "none";
+      gameDifficulty(gameSize);
+      generateRandomCard();
+    } else {
+      alert("Please enter a valid number.");
+    }
+  }
+}
+
+deck.forEach((deckCard) => {
+  for (let i = 0; i < suits.length; i++) {
+    cardsArray.push({ cardName: `${deckCard} of ${suits[i]}`, cardUrl: `./cards/${deckCard + suitsFirstLetter[i]}.webp` });
+  }
+});
+
+function toggleCard(event) {
+  const card = event.target;
+  const cardSrc = card.src.split("/").pop();
+
+  for (let i = 0; i < cardStorageArray.length; i++) {
```

Comment:
You could use [find()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find) instead of a for loop with a return.
```js
const cardFromStorage = cardStorageArray.find(function (c) {
  return c.cardId === card.id;
});
```